feat: contact verification request

richard-ramos opened the pull request:
- [x] Send contact verification request `wakuext_sendContactVerificationRequest(contactID, challenge)`
- [x] Accept contact verification request `wakuext_acceptContactVerificationRequest(contactID, response)`
- [x] Decline contact verification request `wakuext_declineContactVerificationRequest(contactID)`
- [x] Cancel contact verification request `wakuext_cancelVerificationRequest(contactID)`
- [x] Mark as trusted `wakuext_markAsTrusted(contactID)`
- [x] Mark as untrustworthy `wakuext_markAsUntrustworthy(contactID)`
- [x] Remove trust status `wakuext_removeTrustStatus(contactID)`
- [x] Get trust status `wakuext_getTrustStatus(contactID)`
- [ ] Create activity center notifications
- [x] Test units

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (81)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | ae7ca013 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/1/) | 2022-03-14 12:37:01 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220314-123415-ae7ca0-pr2586.zip) |
| :heavy_check_mark: | ae7ca013 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/1/) | 2022-03-14 12:38:24 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220314-123410-ae7ca0-pr2586.aar) |
| :heavy_check_mark: | ae7ca013 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/1/) | 2022-03-14 12:38:32 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220314-123410-ae7ca0-pr2586.zip) |
| :heavy_check_mark: | ae7ca013 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/2/) | 2022-03-14 17:50:41 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220314-174849-ae7ca0-pr2586.zip) |
| :heavy_check_mark: | ae7ca013 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/2/) | 2022-03-14 17:51:50 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220314-174849-ae7ca0-pr2586.aar) |
| :heavy_check_mark: | ae7ca013 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/2/) | 2022-03-14 17:52:14 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220314-174854-ae7ca0-pr2586.zip) |
| | | | | | | |
| :heavy_check_mark: | 9a96bb6f | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/3/) | 2022-03-16 01:21:04 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220316-011747-9a96bb-pr2586.zip) |
| :heavy_check_mark: | 9a96bb6f | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/3/) | 2022-03-16 01:22:06 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220316-011747-9a96bb-pr2586.aar) |
| :heavy_check_mark: | 9a96bb6f | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/3/) | 2022-03-16 01:22:36 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220316-011747-9a96bb-pr2586.zip) |
| | | | | | | |
| :heavy_check_mark: | 1f744c08 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/4/) | 2022-03-16 01:31:14 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220316-012929-1f744c-pr2586.zip) |
| :heavy_check_mark: | 1f744c08 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/4/) | 2022-03-16 01:31:33 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220316-012929-1f744c-pr2586.zip) |
| :heavy_check_mark: | 1f744c08 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/4/) | 2022-03-16 01:32:36 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220316-012929-1f744c-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | af7e5065 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/5/) | 2022-03-16 01:34:18 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220316-013217-af7e50-pr2586.zip) |
| :heavy_check_mark: | af7e5065 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/5/) | 2022-03-16 01:34:18 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220316-013217-af7e50-pr2586.zip) |
| :heavy_check_mark: | af7e5065 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/5/) | 2022-03-16 01:35:33 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220316-013240-af7e50-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | 2bee575e | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/6/) | 2022-03-17 15:43:05 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220317-154126-2bee57-pr2586.zip) |
| :heavy_check_mark: | 2bee575e | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/6/) | 2022-03-17 15:43:34 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220317-154126-2bee57-pr2586.zip) |
| :heavy_check_mark: | 2bee575e | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/6/) | 2022-03-17 15:45:40 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220317-154126-2bee57-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | 36017196 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/7/) | 2022-03-21 16:01:34 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220321-155851-360171-pr2586.zip) |
| :heavy_check_mark: | 36017196 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/7/) | 2022-03-21 16:01:50 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220321-155851-360171-pr2586.zip) |
| :heavy_check_mark: | 36017196 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/7/) | 2022-03-21 16:02:21 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220321-155851-360171-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | 4d67ef36 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/8/) | 2022-03-27 16:26:03 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220327-162410-4d67ef-pr2586.zip) |
| :heavy_check_mark: | 4d67ef36 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/8/) | 2022-03-27 16:27:51 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220327-162410-4d67ef-pr2586.aar) |
| :heavy_check_mark: | 4d67ef36 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/8/) | 2022-03-27 16:28:44 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220327-162410-4d67ef-pr2586.zip) |
| | | | | | | |
| :heavy_check_mark: | d215c4dc | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/9/) | 2022-04-04 02:51:24 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220404-024944-d215c4-pr2586.zip) |
| :heavy_check_mark: | d215c4dc | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/9/) | 2022-04-04 02:52:26 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220404-024941-d215c4-pr2586.zip) |
| :heavy_check_mark: | d215c4dc | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/9/) | 2022-04-04 02:53:12 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220404-024941-d215c4-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | cbd40a01 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/10/) | 2022-04-04 15:33:29 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220404-153156-cbd40a-pr2586.zip) |
| :heavy_check_mark: | cbd40a01 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/10/) | 2022-04-04 15:33:44 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220404-153156-cbd40a-pr2586.zip) |
| :heavy_check_mark: | cbd40a01 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/10/) | 2022-04-04 15:35:28 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220404-153156-cbd40a-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | 88fddb37 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/11/) | 2022-04-06 19:02:30 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220406-185918-88fddb-pr2586.zip) |
| :heavy_check_mark: | 88fddb37 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/11/) | 2022-04-06 19:04:03 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220406-185918-88fddb-pr2586.aar) |
| :heavy_check_mark: | 88fddb37 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/11/) | 2022-04-06 19:04:14 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220406-185918-88fddb-pr2586.zip) |
| | | | | | | |
| :heavy_check_mark: | 165547a6 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/12/) | 2022-04-06 21:09:36 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220406-210757-165547-pr2586.zip) |
| :heavy_check_mark: | 165547a6 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/12/) | 2022-04-06 21:10:01 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220406-210757-165547-pr2586.zip) |
| :heavy_check_mark: | 165547a6 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/12/) | 2022-04-06 21:11:04 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220406-210756-165547-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | afba7fe3 | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/13/) | 2022-04-08 19:12:51 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220408-191058-afba7f-pr2586.zip) |
| :heavy_check_mark: | afba7fe3 | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/13/) | 2022-04-08 19:13:11 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220408-191055-afba7f-pr2586.zip) |
| :heavy_check_mark: | afba7fe3 | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/13/) | 2022-04-08 19:14:20 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220408-191055-afba7f-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | 26e5f1e1 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/14/) | 2022-04-08 20:14:13 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220408-201221-26e5f1-pr2586.zip) |
| :heavy_check_mark: | 26e5f1e1 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/14/) | 2022-04-08 20:17:05 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220408-201221-26e5f1-pr2586.zip) |
| :heavy_check_mark: | 26e5f1e1 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/14/) | 2022-04-08 20:17:16 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220408-201221-26e5f1-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | a6f60a48 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/15/) | 2022-04-08 20:28:38 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220408-202042-a6f60a-pr2586.zip) |
| :heavy_check_mark: | a6f60a48 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/15/) | 2022-04-08 20:29:51 | ~9 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220408-202042-a6f60a-pr2586.aar) |
| :heavy_check_mark: | a6f60a48 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/15/) | 2022-04-08 20:34:05 | ~13 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220408-202042-a6f60a-pr2586.zip) |
| | | | | | | |
| :heavy_check_mark: | c1c71c29 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/16/) | 2022-04-11 17:51:17 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220411-174852-c1c71c-pr2586.zip) |
| :heavy_check_mark: | c1c71c29 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/16/) | 2022-04-11 17:51:23 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220411-174850-c1c71c-pr2586.zip) |
| :heavy_check_mark: | c1c71c29 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/16/) | 2022-04-11 17:52:19 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220411-174850-c1c71c-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | 1ae62083 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/17/) | 2022-04-13 14:17:49 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220413-141602-1ae620-pr2586.zip) |
| :heavy_check_mark: | 1ae62083 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/17/) | 2022-04-13 14:18:26 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220413-141602-1ae620-pr2586.zip) |
| :heavy_check_mark: | 1ae62083 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/17/) | 2022-04-13 14:20:17 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220413-141600-1ae620-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | c1f75113 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/18/) | 2022-04-13 15:43:35 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220413-154144-c1f751-pr2586.zip) |
| :heavy_check_mark: | c1f75113 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/18/) | 2022-04-13 15:44:36 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220413-154143-c1f751-pr2586.zip) |
| :heavy_check_mark: | c1f75113 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/18/) | 2022-04-13 15:46:03 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220413-154141-c1f751-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | dd4c0724 | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/19/) | 2022-04-14 15:23:03 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220414-152050-dd4c07-pr2586.zip) |
| :heavy_check_mark: | dd4c0724 | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/19/) | 2022-04-14 15:23:10 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220414-152053-dd4c07-pr2586.zip) |
| :heavy_check_mark: | dd4c0724 | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/19/) | 2022-04-14 15:24:46 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220414-152050-dd4c07-pr2586.aar) |
| | | | | | | |
| :x: | 21e82af4 | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/20/) | 2022-05-13 15:58:56 | ~44 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/20/consoleText) |
| :x: | 21e82af4 | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/20/) | 2022-05-13 15:59:42 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/20/consoleText) |
| :x: | 21e82af4 | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/20/) | 2022-05-13 16:00:08 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/20/consoleText) |
| | | | | | | |
| :x: | cef9178e | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/21/) | 2022-05-13 19:03:59 | ~19 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/21/consoleText) |
| :x: | cef9178e | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/21/) | 2022-05-13 19:04:15 | ~36 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/21/consoleText) |
| :x: | cef9178e | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/21/) | 2022-05-13 19:04:49 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/21/consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3b4d2649 | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/22/) | 2022-05-13 19:19:29 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220513-191728-3b4d26-pr2586.zip) |
| :heavy_check_mark: | 3b4d2649 | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/22/) | 2022-05-13 19:20:45 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220513-191727-3b4d26-pr2586.zip) |
| :heavy_check_mark: | 3b4d2649 | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/22/) | 2022-05-13 19:21:15 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220513-191727-3b4d26-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | 80338ced | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/23/) | 2022-06-15 16:36:06 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220615-163350-80338c-pr2586.zip) |
| :heavy_check_mark: | 80338ced | [#23](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/23/) | 2022-06-15 16:38:34 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220615-163349-80338c-pr2586.zip) |
| :heavy_check_mark: | 80338ced | [#23](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/23/) | 2022-06-15 16:38:35 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220615-163349-80338c-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | d004526b | [#24](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/24/) | 2022-06-23 20:02:40 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220623-200008-d00452-pr2586.zip) |
| :heavy_check_mark: | d004526b | [#24](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/24/) | 2022-06-23 20:04:27 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220623-200007-d00452-pr2586.zip) |
| :heavy_check_mark: | d004526b | [#24](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/24/) | 2022-06-23 20:04:52 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220623-200005-d00452-pr2586.aar) |
| | | | | | | |
| :heavy_check_mark: | 9a8c27d7 | [#25](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/25/) | 2022-06-27 19:26:32 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220627-192445-9a8c27-pr2586.zip) |
| :heavy_check_mark: | 9a8c27d7 | [#25](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/25/) | 2022-06-27 19:27:59 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220627-192442-9a8c27-pr2586.aar) |
| :heavy_check_mark: | 9a8c27d7 | [#25](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/25/) | 2022-06-27 19:28:27 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220627-192445-9a8c27-pr2586.zip) |
| | | | | | | |
| :heavy_check_mark: | 9f66c938 | [#26](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/26/) | 2022-06-29 17:03:54 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220629-165958-9f66c9-pr2586.zip) |
| :heavy_check_mark: | 9f66c938 | [#26](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/26/) | 2022-06-29 17:04:45 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220629-165957-9f66c9-pr2586.aar) |
| :heavy_check_mark: | 9f66c938 | [#26](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/26/) | 2022-06-29 17:06:57 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220629-170002-9f66c9-pr2586.zip) |
| | | | | | | |
| :heavy_check_mark: | 2a1f3cfd | [#27](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/27/) | 2022-06-30 14:48:57 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220630-144536-2a1f3c-pr2586.aar) |
| :heavy_check_mark: | 2a1f3cfd | [#27](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/27/) | 2022-06-30 14:49:28 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220630-144536-2a1f3c-pr2586.zip) |
| :heavy_check_mark: | 2a1f3cfd | [#27](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/27/) | 2022-06-30 14:51:19 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220630-144540-2a1f3c-pr2586.zip) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 50e07798 | [#28](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/28/) | 2022-07-04 18:26:37 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220704-182429-50e077-pr2586.zip) |
| :heavy_check_mark: | 50e07798 | [#28](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/28/) | 2022-07-04 18:28:21 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220704-182428-50e077-pr2586.aar) |
| :heavy_check_mark: | 50e07798 | [#28](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/28/) | 2022-07-04 18:28:32 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220704-182428-50e077-pr2586.zip) |
| | | | | | | |
| :heavy_check_mark: | b2e501f4 | [#29](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2586/29/) | 2022-07-05 19:42:33 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220705-194055-b2e501-pr2586.zip) |
| :heavy_check_mark: | b2e501f4 | [#29](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2586/29/) | 2022-07-05 19:43:27 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220705-194055-b2e501-pr2586.zip) |
| :heavy_check_mark: | b2e501f4 | [#29](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2586/29/) | 2022-07-05 19:44:04 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220705-194054-b2e501-pr2586.aar) |

Samyoul commented on protocol/contact.go:
```diff
@@ -55,6 +56,14 @@ func (c *Contact) CanonicalImage(profilePicturesVisibility accounts.ProfilePictu
 
+type VerificationStatus int
+
+const (
+	VerificationStatusUNVERIFIED VerificationStatus = 0
+	VerificationStatusVERIFYING  VerificationStatus = 1
+	VerificationStatusVERIFIED   VerificationStatus = 2
+)
```
Would you consider using `iota`, it seems like a nice fit for your enums

```suggestion
const (
	VerificationStatusUNVERIFIED VerificationStatus = iota
	VerificationStatusVERIFYING  VerificationStatus
	VerificationStatusVERIFIED   VerificationStatus
)
```

Samyoul commented on protocol/verification/persistence.go:
```diff
@@ -0,0 +1,252 @@
+		&repliedAt,
+	)
+
+	switch err {
+	case sql.ErrNoRows:
+		return nil, nil
+	case nil:
+		vr.RequestedAt = time.Unix(requestedAt, 0)
+		vr.RepliedAt = time.Unix(repliedAt, 0)
+		return &vr, nil
+	default:
+		return nil, err
+	}
```
I like this

Samyoul commented on protocol/verification/persistence.go:
```diff
@@ -0,0 +1,252 @@
+	}
+
+	if numRows == 0 {
+		return errors.New("verification request not found")
```
Would you consider something like? If you've got multiple instances of the same error being returned, consuming code can check for a specific err via direct comparison `if err == ErrVerificationRequestNotFound { do something }`, also it saves you time if you need to call the same error in other places.

Obviously the `var()` block would go at the top in the package context.

```suggestion
var (
	ErrVerificationRequestNotFound = errors.New("verification request not found")
)

		return ErrVerificationRequestNotFound 
```

Samyoul reviewed (approved):
Great PR, a few points but just convention stuff.

jrainville commented on protocol/messenger_contact_verification.go:
```diff
@@ -0,0 +1,726 @@
+
+	state.AllVerificationRequests = append(state.AllVerificationRequests, persistedVR)
+
+	contact.VerificationStatus = VerificationStatusVERIFIED
```
I have a small issue where at the end of the process, the request is marked as trusted, but the verificationStatus stays at 0 when I look at it on the client side, maybe you'll be able to help me @richard-ramos

0x-r4bbit replied:
Is this still relevant?

jrainville replied:
Ah my bad, Richard helped me fix that

jrainville committed:
feat: contact verification request

fix: add verification request to response

fix: code review

add missing functions and simplify timestamp usage

fix: sync verification requests

feat: add endpoint to fetch all received verification requests

feat: add signal when trusting verification request

churik commented:
@richard-ramos can you please resolve conflicts first and rebase to current develop? 
Thanks!

jrainville commented:
This can be reviewed and merged when done

0x-r4bbit commented on protocol/messenger.go:
```diff
@@ -5723,3 +5897,79 @@ func (m *Messenger) handleSyncClearHistory(state *ReceivedMessageState, message
+	return nil
+}
+
+func ToVerificationRequest(message protobuf.SyncVerificationRequest) *verification.Request {
```
Would be cool if this went into `protocol/requests/...` instead for consistency's sake, but doesn't have to

0x-r4bbit commented on protocol/messenger.go:
```diff
@@ -5723,3 +5897,79 @@ func (m *Messenger) handleSyncClearHistory(state *ReceivedMessageState, message
+		m.allContacts.Store(contact.ID, contact)
+		state.ModifiedContacts.Store(contact.ID, true)
+
+		// TODO: create activity center notif
```
Are these yet to be done as part of this PR or is has this been put on TODO on purpose for now?

jrainville replied:
Activity center stuff will be done in another PR

0x-r4bbit commented on protocol/migrations/sqlite/1655309635_contact_verification.up.sql:
```diff
@@ -0,0 +1,20 @@
+  replied_at INT NOT NULL DEFAULT 0,
+  verification_status INT NOT NULL DEFAULT 0,
+  CONSTRAINT fromto_unique UNIQUE (from_user, to_user) ON CONFLICT REPLACE
+);
```
Need to ensure we update the timestamp of this file right before merging (and then `make generate` again)

jrainville replied:
Done

0x-r4bbit reviewed (commented):
Massive work!!
Left some tiny comments, but really well done (also thanks for adding all those tests!)
